pyhanko/sign/fields.py

```diff
@@ -1665,6 +1665,23 @@ def append_signature_field(
 
     if sig_field_spec.box is not None:
         llx, lly, urx, ury = sig_field_spec.box
+        matrix = None
+        if sig_field_spec.visible_sig_settings.rotate_with_page == False:
+            pagetree_obj = page_ref.get_object()
+            while '/Parent' in pagetree_obj and ('/Type' not in pagetree_obj or pagetree_obj['/Type'] != '/Page'):
```

Comment:
That conditional looks doesn't look quite right. You want the closest ancestor with a `/Rotate` entry, no?

Also, perhaps keep track of seen pages here to prevent loops :).

Reply:
Rotating the signature would only make sense if the page itself is rotated. Does pyHanko even add the signature to anything other than the page itself?

Im not sure if a `/Parent` field can ever become a loop, unless someone didnt understand what a parent is supposed to be in programming

Reply:
> Rotating the signature would only make sense if the page itself is rotated. Does pyHanko even add the signature to anything other than the page itself?

I thought you were trying to look for the closest ancestor with a `/Rotate` flag (that would be inherited down to the leaf node) to check if the page you're interested in was rotated? Or did I misinterpret your intent here?

> Im not sure if a /Parent field can ever become a loop, unless someone didnt understand what a parent is supposed to be in programming

Of course it can't be a loop in a well-formed PDF file, but getting stuck in an infinite loop on malformed ones is also not something we want ;).

Reply:
> I thought you were trying to look for the closest ancestor with a `/Rotate` flag (that would be inherited down to the leaf node) to check if the page you're interested in was rotated? Or did I misinterpret your intent here?

No, you are totally right. I just assumed that there would be no rotation between signature and the actual page based on how pyHanko places the signature in the pdf. I will change it as soon as i find the time

> Of course it can't be a loop in a well-formed PDF file, but getting stuck in an infinite loop on malformed ones is also not something we want ;).

Who doesnt like infinite loops? You do have a point, you can never know how PDFs are generated. The reason i even opened the issue was because i had to deal with a PDF that was generated in portrait orientation and then rotated instead of directly generating it in landscape orientation